tests/integration/test_coverage_ratchet_gaps.py

```
@@ -0,0 +1,1502 @@
+# ===========================================================================
+
+
+class TestBufferPoolGaps:
+    def test_buffer_pool_init_validation(self) -> None:
+        from file_organizer.optimization.buffer_pool import BufferPool
+
+        with pytest.raises(ValueError, match="buffer_size must be > 0"):
+            BufferPool(buffer_size=0)
+        with pytest.raises(ValueError, match="initial_buffers must be > 0"):
+            BufferPool(initial_buffers=0)
+        with pytest.raises(ValueError, match="max_buffers .* must be >= initial_buffers"):
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)